In [1]:
p=3.14
r=float(input("Enter radius of cylinder: "))
h=float(input("Enter height of cylinder: "))
sur_area=2*p*r*(h+r)
vol=p*r*r*h
print("Surface area of cylinder: ",sur_area, "\nVolume of cylinder: ",vol)

Enter radius of cylinder:  3
Enter height of cylinder:  4


Surface area of cylinder:  131.88 
Volume of cylinder:  113.03999999999999


In [1]:
p=float(input("Enter principle amount: "))
n=float(input("Enter no. of years: "))
r=float(input("Enter rate of interest: "))
print("-------------------------------",
      "\nPrinciple amount: ",p,
      "\nNo. of years: ",n,
      "\nInterest rate: ",r,
      "\nSimple Interest: ",(p*n*r)/100)

Enter principle amount:  564
Enter no. of years:  4
Enter rate of interest:  11


------------------------------- 
Principle amount:  564.0 
No. of years:  4.0 
Interest rate:  11.0 
Simple Interest:  248.16


In [2]:
#area if triangle by Heron's Formula
a=float(input("Enter 1st side: "))
b=float(input("Enter 2nd side: "))
c=float(input("Enter 3rd side: "))
s=(a+b+c)/2
area=(s*(s-a)*(s-b)*(s-c))**1/2
print("Area: ",area)

Enter 1st side:  5
Enter 2nd side:  4
Enter 3rd side:  6


Area:  49.21875


In [3]:
#Control Statement
#if...else
num=int(input("Enter the number"))
if(num>=0):
    print("Number is Pos")
else:
    print("Number is Neg")

Enter the number 36


Number is Pos


In [4]:
#Control Statement
#if...else
num=int(input("Enter the number"))
if(num%2==0):
    print("Number is Even")
else:
    print("Number is Odd")

Enter the number 36


Number is Even


In [5]:
#Control Statement
#if...else
num=int(input("Enter the number"))
if(num%2!=0):
    print("Number is Odd")
else:
    print("Number is Even")

Enter the number 35


Number is Odd


In [6]:
#Control Statement
#if...else
a=int(input("Enter the value of a"))
b=int(input("Enter the value of b"))
if(a>b):
    print("A is greater than B")
else:
    print("B is greater than A")

Enter the value of a 56
Enter the value of b 65


B is greater than A


In [7]:
#Control Statement
#if...else
a=int(input("Enter the value of a"))
b=int(input("Enter the value of b"))
if(a>b):
    print(a," is greater than ",b)
else:
    print(b," is greater than ",a)

Enter the value of a 56
Enter the value of b 65


65  is greater than  56


In [8]:
a=int(input("1st num: "))
b=int(input("2nd num: "))
if a>b:
    print(a," is greater than ",b)
elif a==b:
    print(a," is equal to ",b)
else:
    print(a," is smaller than ",b)

1st num:  56
2nd num:  65


56  is smaller than  65


In [9]:
#voting eligibility
a=int(input("Enter your age: "))
if a>18:
    print("Eligible to vote")
else:
    print("Not eligible to vote")

Enter your age:  56


Eligible to vote


In [10]:
#discount condition
a=int(input("Enter amount: "))
if a>500:
    print("Customer eligible for discount")
else:
    print("Customer not eligible for discount")

Enter amount:  65


Customer not eligible for discount


In [11]:
# Correct credentials
correct_username = "admin"
correct_password = "12345"

# Ask user for input
username = input("Enter Username: ")
password = input("Enter Password: ")

# Check login
if username == correct_username and password == correct_password:
    print("Login successful ✅")
else:
    print("Invalid login ❌")

Enter Username:  admin 
Enter Password:  12345


Invalid login ❌


In [12]:
#Nesting if..else
a=int(input("Ente the value of a"))
b=int(input("Ente the value of b"))
c=int(input("Ente the value of c"))
if(a>b):
    if(a>c):
        print("A is greater than B and C")
    else:
        print("C is greater than A and B")
else:
    if(b>c):
        print("B is greater than A and C")
    else:
        print("C is greater than A and B")

Ente the value of a 3
Ente the value of b 4
Ente the value of c 5


C is greater than A and B


In [13]:
#Nesting if..else
a=int(input("Ente the value of a"))
b=int(input("Ente the value of b"))
c=int(input("Ente the value of c"))
if(a>b):
    if(a>c):
        print("A is greater than B and C")
    else:
        print("C is greater than A and B")
elif(b>c):
    print("B is greater than A and C")
else:
    print("C is greater than A and B")

Ente the value of a 3
Ente the value of b 4
Ente the value of c 5


C is greater than A and B


In [14]:
a=int(input("Enter marks"))
if a>=40:
    if a>=75:
        print("Distinction")
    else:
        print("Just Passed")
else:
    print("Fail")

Enter marks 78


Distinction


In [15]:
acc_bal=50000
dai_lim=10000
amount=int(input("Enter amount to withdraw"))
if amount<=acc_bal:
    if amount<=dai_lim:
        print("Withdraw Successfull")
    else:
        print("Exceed daily limit")
else:
    print("Insufficient Balance")

Enter amount to withdraw 5000


Withdraw Successfull


In [17]:
nationality= "india"
age=int(input("Enter age: "))
country=str(input("Enter country: "))
if age>=18:
    if nationality==country.lower():
        print("Eligible to vote")
    else:
        print("Not eligible (wrong nationality)")
else:
    print("Not eligible (too young to vote)")

Enter age:  19
Enter country:  india


Eligible to vote


In [18]:
signal=input("Enter signal color (red/yellow/green)").lower()
pedestrain=input("Pedstrain button is pressed (yes/no)?").lower()
if signal=='red':
    if pedestrain=='yes':
        print("Wait for vechicles to stop, then cross safely")
    else:
        print("Stop. Do not move")
elif signal=='yellow':
    print("Be ready signal will change soon")
elif signal=='green':
    print("you can go")
else:
    print("Invalid signal color entered")

Enter signal color (red/yellow/green) green
Pedstrain button is pressed (yes/no)? no


you can go


In [19]:
a=int(input("Enter 1st no"))
b=int(input("Enter 2nd no"))
c=int(input("Enter 3rd no"))
if(a>b):
    if(a>c):
        print("A is greater than B and C")
    else:
        print("C is greater than B and A")
elif(b>c):
    print("B is greater than A and C")
else:
    print("C is greater than A and B")

Enter 1st no 56
Enter 2nd no 65
Enter 3rd no 96


C is greater than A and B


In [20]:
m1=int(input("Enter the value of m1"))
m2=int(input("Enter the value of m2"))
m3=int(input("Enter the value of m3"))
m4=int(input("Enter the value of m4"))
total=m1+m2+m3+m4
per=total/4
print("Total",total,"Per=",per)
if(per>=75):
    print("A+")
elif(per>=60):
    print("A")
elif(per>=45):
    print("B")
elif(per>=35):
    print("C")
else:
    print("Fail")

Enter the value of m1 56
Enter the value of m2 89
Enter the value of m3 96
Enter the value of m4 36


Total 277 Per= 69.25
A


Ladder if-else

In [21]:
a=int(input("Enter your age: "))
if a>=60:
    print("Senior")
elif a>=18:
    print("Adult")
elif a>=12:
    print("Teenager")
else:
    print("Child")

Enter your age:  69


Senior


In [22]:
#traffic signal systems
color=str(input("Enter color of signal (red/yellow/green): ")).lower()
if color=='red':
    print("Please stop")
elif color=='yellow':
    print('Ready to stop')
elif color=='green':
    print('You may go')

Enter color of signal (red/yellow/green):  red


Please stop


In [23]:
#number is divisible by 2, 3, 5, or 7
a=int(input("Enter your numeber: "))
if a%2==0:
    print("Number is divisible by 2")
elif a%3==0:
    print("Number is divisible by 3")
elif a%5==0:
    print("Number is divisible by 5")
elif a%7==0:
    print("Number is divisible by 7")
else:
    print("Number is not divisible by 2, 3, 5 and 7 ")

Enter your numeber:  96


Number is divisible by 2


In [24]:
#Fitness level → Beginner, Intermediate, Advanced
p=int(input("Enter pushups capacity"))
c=int(input("Enter crunches capacity"))
s=int(input("Enter squats capacity"))
if p>=75 and c>=50 and s>=100:
    print("Your level is Advanced")
elif p>=25 and c>=20 and s>=40:
    print("Your level is Intermediate")
else:
    print("Your level is Beginner")

Enter pushups capacity 50
Enter crunches capacity 50
Enter squats capacity 50


Your level is Intermediate


In [25]:
#Classify numbers → Prime, Composite, Neither
n = int(input("Enter a number: "))
if n <= 1:
    print("Neither Prime nor Composite")
elif n == 2:
    print("Prime")
else:
    is_prime = True
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            is_prime = False
            break

    if is_prime:
        print("Prime")
    else:
        print("Composite")

Enter a number:  45


Composite


In [26]:
i=1
while(i<=10):
    print(i)
    i=i+1

1
2
3
4
5
6
7
8
9
10


In [27]:
i=10
while(i>=1):
    print(i)
    i=i-1

10
9
8
7
6
5
4
3
2
1


In [28]:
i=-10
while(i<=-1):
    print(i)
    i=i+1

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


In [29]:
i=-1
while(i>=-10):
    print(i)
    i=i-1

-1
-2
-3
-4
-5
-6
-7
-8
-9
-10


In [30]:
i=1
while(i<11):
    print(i)
    i+=1

1
2
3
4
5
6
7
8
9
10


In [31]:
i=10
while(i>0):
    print(i)
    i-=1

10
9
8
7
6
5
4
3
2
1


In [32]:
i=-1
while(i>-11):
    print(i)
    i-=1

-1
-2
-3
-4
-5
-6
-7
-8
-9
-10


In [33]:
i=-10
while(i<-0):
    print(i)
    i+=1

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


In [34]:
i=1
total=0
while(i<=5):
    total+=i
    i+=1
print("Sum: ",total)

Sum:  15


In [35]:
i=1
n=int(input("Enter ending number"))
total=0
while(i<=n):
    total+=i
    i+=1
print("Sum: ",total)

Enter ending number 10


Sum:  55


In [36]:
i=int(input("Enter beginning number"))
n=int(input("Enter ending number"))
total=0
while(i<=n):
    total+=i
    i+=1
print("Sum: ",total)

Enter beginning number 2
Enter ending number 8


Sum:  35


In [37]:
i=1
while(i<=10):
    print("3 * ",i," = ",3*i)
    i+=1

3 *  1  =  3
3 *  2  =  6
3 *  3  =  9
3 *  4  =  12
3 *  5  =  15
3 *  6  =  18
3 *  7  =  21
3 *  8  =  24
3 *  9  =  27
3 *  10  =  30


In [38]:
n=int(input("Enter desired number"))
i=1
while(i<=10):
    print(n," * ",i," = ",n*i)
    i+=1

Enter desired number 6


6  *  1  =  6
6  *  2  =  12
6  *  3  =  18
6  *  4  =  24
6  *  5  =  30
6  *  6  =  36
6  *  7  =  42
6  *  8  =  48
6  *  9  =  54
6  *  10  =  60


In [39]:
#reverse number
n = 1234
rev = 0
while n > 0:
    digit = n % 10
    rev = rev * 10 + digit
    n //= 10

print("Reversed Number:", rev)

Reversed Number: 4321


Dry Run (Step by Step)

First iteration:

digit = 1234 % 10 = 4

rev = 0 * 10 + 4 = 4

n = 1234 // 10 = 123

Second iteration:

digit = 123 % 10 = 3

rev = 4 * 10 + 3 = 43

n = 123 // 10 = 12

Third iteration:

digit = 12 % 10 = 2

rev = 43 * 10 + 2 = 432

n = 12 // 10 = 1

Fourth iteration:

digit = 1 % 10 = 1

rev = 432 * 10 + 1 = 4321

n = 1 // 10 = 0

In [95]:
n=int(input("Enter number: "))
rev=0
while n>0:
    digit=n%10
    rev=rev*10+digit
    n//=10
print(rev)

Enter number:  123456


654321


In [40]:
for i in range(10):
    print(i)

0
1
2
3
4
5
6
7
8
9


In [41]:
for i in range(1,11):
    print(i)

1
2
3
4
5
6
7
8
9
10


In [42]:
for i in range(10,0,-1):
    print(i)

10
9
8
7
6
5
4
3
2
1


In [43]:
for i in range(-10,0,1):
    print(i)

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


In [44]:
for i in range(-1,-11,-1):
    print(i)

-1
-2
-3
-4
-5
-6
-7
-8
-9
-10


In [45]:
word='python'
for ch in word:
    print(ch)

p
y
t
h
o
n


In [46]:
total=0
for i in range(11):
    total+=i
print(total)

55


In [47]:
n=int(input('Enter desired no'))
total=0
for i in range(n):
    total+=i
print("total: ",total)

Enter desired no 11


total:  55


In [48]:
num=3
for i in range(1,11):
    print(num," * ",i," = ",num*i)

3  *  1  =  3
3  *  2  =  6
3  *  3  =  9
3  *  4  =  12
3  *  5  =  15
3  *  6  =  18
3  *  7  =  21
3  *  8  =  24
3  *  9  =  27
3  *  10  =  30


In [49]:
num=int(input('Enter desired no: '))
for i in range(1,11):
    print(num," * ",i," = ",num*i)

Enter desired no:  11


11  *  1  =  11
11  *  2  =  22
11  *  3  =  33
11  *  4  =  44
11  *  5  =  55
11  *  6  =  66
11  *  7  =  77
11  *  8  =  88
11  *  9  =  99
11  *  10  =  110


In [50]:
for i in range(2,20,5):
    print(i)

2
7
12
17


In [51]:
for i in range(1,11):
    print("Square of ",i,"= ",i*i)

Square of  1 =  1
Square of  2 =  4
Square of  3 =  9
Square of  4 =  16
Square of  5 =  25
Square of  6 =  36
Square of  7 =  49
Square of  8 =  64
Square of  9 =  81
Square of  10 =  100


In [52]:
n=int(input('Enter desired no: '))
for i in range(1,n+1):
    print("Square of ",i,"= ",i*i)

Enter desired no:  11


Square of  1 =  1
Square of  2 =  4
Square of  3 =  9
Square of  4 =  16
Square of  5 =  25
Square of  6 =  36
Square of  7 =  49
Square of  8 =  64
Square of  9 =  81
Square of  10 =  100
Square of  11 =  121


In [53]:
word = "hello"
rev = ""
for ch in word:
    rev = ch + rev
print("Reversed =", rev)

Reversed = olleh


In [54]:
word = input('Enter desired word: ')
rev = ""
for ch in word:
    rev = ch + rev
print("Reversed =", rev)

Enter desired word:  abcdefghi


Reversed = ihgfedcba


In [55]:
names=['Hashirama','Tobirama','Naruto']
for i in range(len(names)):
    print("index: ",i,"values: ", names[i])

index:  0 values:  Hashirama
index:  1 values:  Tobirama
index:  2 values:  Naruto


In [56]:
#vowels counting
text='pogramming'
count=0
for ch in text:
    if ch in 'aeiou':
        count+=1
print('total vowels :',count)

total vowels : 3


In [57]:
text=str(input('Enter desired word: '))
count=0
for ch in text:
    if ch in 'aeiou':
        count+=1
print('total vowels :',count)

Enter desired word:  encyclopedia


total vowels : 5


In [97]:
text='polytetradifluoroethene'
count=0
for ch in text:
    if ch in 'aeiou':
        count+=1
print(count)

10


In [1]:
text="KatenKyokutsuKaramatsuShinju"
count=0
for ch in text:
    if ch in 'aeiou':
        count+=1
print("Total:", count)

Total: 11


In [58]:
#factorial
num=5
fact=1
for i in range(1,num+1):
    fact*=i
print(fact)

120


In [59]:
num=int(input('Enter desired no: '))
fact=1
for i in range(1,num+1):
    fact*=i
print(fact)

Enter desired no:  10


3628800


In [60]:
student={"name": "Ravi", "age": 20, "marks": 85}
for key in student:
    print("Key ",key, "Elements ",student[key])

Key  name Elements  Ravi
Key  age Elements  20
Key  marks Elements  85


In [61]:
names={"name": "Ravi", "age": 20, "marks": 85}
for key in names:
    print('key ',key,'values', names[key])

key  name values Ravi
key  age values 20
key  marks values 85


In [62]:
#unicode for each character
for ch in "ABCXYZ123":
    print(ch, "->", ord(ch))

A -> 65
B -> 66
C -> 67
X -> 88
Y -> 89
Z -> 90
1 -> 49
2 -> 50
3 -> 51


In [63]:
#cube
for i in range(1,6):
    print("Cube of ",i," is",i**3)

Cube of  1  is 1
Cube of  2  is 8
Cube of  3  is 27
Cube of  4  is 64
Cube of  5  is 125


In [64]:
#odd num
for i in range(1, 21, 2):
    print(i)


1
3
5
7
9
11
13
15
17
19


In [65]:
#consonants
text='programming'
count=0
for ch in text:
    if ch not in 'aeiou':
        count+=1
count

8

In [66]:
#consonants counting
text=str(input("Enter word: "))
count=0
for ch in text:
    if ch not in 'aeiou':
        count+=1
count

Enter word:  encyclopedia


7

In [67]:
numbers = [12, 45, 7, 89, 34]
largest=numbers[0]
for num in numbers:
    if num>largest:
        largest=num
print("Largest number: ",largest)

Largest number:  89


In [68]:
# Ask user how many numbers
n = int(input("Enter how many numbers: "))

numbers = []
for i in range(n):
    num = int(input("Enter number: "))
    numbers.append(num)

# Assume first number is largest
largest = numbers[0]

# Check each number
for num in numbers:
    if num > largest:
        largest = num

print("Largest number =", largest)

Enter how many numbers:  3
Enter number:  11
Enter number:  12
Enter number:  13


Largest number = 13


In [71]:
n=int(input("Enter how many numbers: "))

numbers=[]
for i in range(n):
    num=int(input("Enter number: "))
    numbers.append(num)

largest=numbers[0]
for num in numbers:
    if num>largest:
        largest=num

print(largest)

Enter how many numbers:  3
Enter number:  11
Enter number:  12
Enter number:  13


13


In [72]:
colors = ["red", "green", "blue","black","white"]
for index, value in enumerate(colors):
    print(index, "->", value)

0 -> red
1 -> green
2 -> blue
3 -> black
4 -> white


In [73]:
#Sum of all even numbers in a list
nums = [10, 21, 32, 43, 54]
total = 0
for n in nums:
    if n%2==0:
        total+=n
print(total)

96


In [74]:
#Reverse a list using for loop
lst = [1, 2, 3, 4, 5]
rev = []
for i in lst:
    rev = [i] + rev
print("Reversed list =", rev)

Reversed list = [5, 4, 3, 2, 1]


In [2]:
lst=[11,22,33,44,55]
rev=[]
for i in lst:
    rev=[i]+rev
print(rev)

[55, 44, 33, 22, 11]


In [75]:
#print each word 
sentence = "Python is fun"
for word in sentence.split():
    print(word)

Python
is
fun


In [76]:
#numbers with their squares in dictionary form
squares = {}
for i in range(1, 6):
    squares[i] = i * i
print(squares)

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25}


In [12]:
value=123.45678
width=10
precision=3
f"the value is{value:{width}.{precision}}"

'the value is  1.23e+02'

In [78]:
value=123.45678
width=10
precision=7
f"the value is {value:{width}.{precision}}"

'the value is   123.4568'

In [91]:
#if-else operations
import math
operation= ['+','-','/','*','sin','cos','tan','sqrt','log','power']

choice= input("Enter operation ('+','-','/','*','sin','cos','tan','sqrt','log','power'): ")

if choice in ['+','-','/','*']:
    x=float(input("Enter first number"))
    y=float(input("Enter second number"))
    if choice=='+':
        print('x+y: ',x+y)
    elif choice=='-':
        print('x-y: ',x-y)
    elif choice=='*':
        print('x*y: ',x*y)
    elif choice=='-':
        print('x/y: ',x/y)
elif choice=='sin':
    angle=math.radians(float(input('Enter angle in degrees: ')))
    print('sin: ',math.sin(angle))
elif choice=='cos':
    angle=math.radians(float(input('Enter angle in degrees: ')))
    print('cos: ',math.cos(angle))
elif choice=='tan':
    angle=math.radians(float(input('Enter angle in degrees: ')))
    print('tan:',math.tan(angle))
elif choice=='sqrt':
    num=float(input('Enter number: '))
    print('Square root: ',math.sqrt(num))
elif choice=='log':
    num1=float(input('Enter number: '))
    print('log(base e): ',math.log(num))
elif choice=='power':
    x=float(input("Enter first number"))
    y=float(input("Enter second number"))
    print('Power',math.pow(x,y))
else:
    print("Incorrect Operation")

Enter operation ('+','-','/','*','sin','cos','tan','sqrt','log','power'):  sin
Enter angle in degrees:  30


sin:  0.49999999999999994


In [9]:
#if-else operations
import math
operation= ['+','-','/','*','sin','cos','tan','sqrt','log','power']

choice= input("Enter operation ('+','-','/','*','sin','cos','tan','sqrt','log','power'): ")

if choice in ['+','-','/','*']:
    x=int(input("Enter value x: "))
    y=int(input("Enter value y: "))
    if choice=='+':
        print("Addition x+y",x+y)
    elif choice=='-':
        print("Subtraction x-y",x-y)
    elif choice=='*':
        print("Multiplication x*y",x*y)
    elif choice=='/':
        print("Division x/y")
elif choice in ['sin','cos','tan']:
    angle=math.radians(float(input("Enter angle: ")))
    if choice=='sin':
        print(f"Sin:{math.sin(angle)}")
    elif choice=='cos':
        print(f"Cos:{math.cos(angle)}")
    elif choice=='tan':
        print(f"Tan:{math.tan(angle)}")
elif choice=='sqrt':
    value=int(input("Enter value:"))
    print(f"Sq. root of {value}:{math.sqrt(value)}")
elif choice=='log':
    value=int(input("Enter value:"))
    print(f"Log of {value} to base e {math.log(value)}")
elif choice=='power':
    base=int(input("Enter value base: "))
    exp=int(input("Enter value exponential: "))
    print("Power ",math.pow(base,exp))
else:
    print("Incorrect Operation")

Enter operation ('+','-','/','*','sin','cos','tan','sqrt','log','power'):  gfthdf


Incorrect Operation


In [92]:
import sympy as sp

x = sp.symbols('x')
f = x**3 + 2*x**2 + 5

df = sp.diff(f, x)
print("Derivative:", df)

integral = sp.integrate(f, x)
print("Integral:", integral)

def_integral = sp.integrate(f, (x, 0, 2))
print("Definite Integral (0 to 2):", def_integral)

Derivative: 3*x**2 + 4*x
Integral: x**4/4 + 2*x**3/3 + 5*x
Definite Integral (0 to 2): 58/3


In [99]:
correct_username='luffy'
correct_password='12345'

username=input("Enter username: ")
password=input("Enter password: ")

if username==correct_username and password==correct_password:
    print("Login succesfull")
else: 
    print("Invalid login")

Enter username:  luffy
Enter password:  12345


Login succesfull


In [105]:
#permutation and combination
import math
n=int(input("Enter value for n: "))
r=int(input("Enter value for r: "))

nPr=math.factorial(n)/math.factorial(n-r)
nCr=math.factorial(n)/(math.factorial(r)*math.factorial(n-r))

print("Permuation: ",nPr,"\nCombination: ",nCr)

Enter value for n:  5
Enter value for r:  2


Permuation:  20.0 
Combination:  10.0


In [11]:
import math
a,b,c=1,-5,6

d=b**2-4*a*c    #b^2-4ac  Discriminant

if d>0:
    root1=-b+math.sqrt(d)/(2*a)
    root2=-b-math.sqrt(d)/(2*a)
    print(f"Root1: {root1} Root2: {root2}")
elif d==0:
    root=-(b/2*a)
    print("Root: ",root)
else:
    real=-(b/2*a)
    imag=math.sqrt(d)/(2*a)
    print(f"{real}+{imag}i and {real}-{imag}i")

Root1: 5.5 Root2: 4.5


In [110]:
import math
a=int(input("Enter a: "))
b=int(input("Enter b: "))
c=int(input("Enter c: "))

d=b**2-4*a*c    #b^2-4ac  Discriminant

if d>0:
    root1=(-b+math.sqrt(d))/(2*a)
    root2=(-b-math.sqrt(d))/(2*a)
    print(f"Root1: {root1} Root2: {root2}")
elif d==0:
    root=-(b/2*a)
    print("Root: ",root)
else:
    real=-(b/2*a)
    imag=math.sqrt(-d)/(2*a)
    print("Complex roots: "f"{real} + {imag}i and {real} - {imag}i")

Enter a:  3
Enter b:  4
Enter c:  5


Complex roots: -6.0 + 1.1055415967851332i and -6.0 - 1.1055415967851332i


In [2]:
year=int(input("Enter year: "))
print(f"{year} is a Leap year") if(year%400==0 or (year%4==0 and year%100!=0)) else print(f"{year} is not a leap year")

Enter year:  2356


2356 is a Leap year


In [15]:
year=int(input("Enter year: "))
print(f"{year} is a leap year") if((year%400==0) or (year%100!=0 or year%4==0)) else print(f"{year} is not a leap year")

Enter year:  2400


2400 is a leap year


In [6]:
a=1
s=0
while a<=50:
    print(a,end=" ")
    s=s+a
    a=a+1
print('\nSum',s)

1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 
Sum 1275


In [14]:
a=1
while a<=100:
    print(a,end=" ")
    if a%10==0:
        print()
    a=a+1

1 2 3 4 5 6 7 8 9 10 
11 12 13 14 15 16 17 18 19 20 
21 22 23 24 25 26 27 28 29 30 
31 32 33 34 35 36 37 38 39 40 
41 42 43 44 45 46 47 48 49 50 
51 52 53 54 55 56 57 58 59 60 
61 62 63 64 65 66 67 68 69 70 
71 72 73 74 75 76 77 78 79 80 
81 82 83 84 85 86 87 88 89 90 
91 92 93 94 95 96 97 98 99 100 


In [11]:
num=1
sum_result=0
while num<=5:
    sum_result+=num
    num+=1
else:
    print("Loop completed sum=",sum_result)

Loop completed sum= 15


In [16]:
i=1
while i<=4:
    j=1
    while j<=2:
        print(f"Outer loop: {i} Inner loop: {j}")
        j+=1
    i+=1

Outer loop: 1 Inner loop: 1
Outer loop: 1 Inner loop: 2
Outer loop: 2 Inner loop: 1
Outer loop: 2 Inner loop: 2
Outer loop: 3 Inner loop: 1
Outer loop: 3 Inner loop: 2
Outer loop: 4 Inner loop: 1
Outer loop: 4 Inner loop: 2


In [17]:
list=['Naruto','Sasuke','Hinata','Ino']
for index in range(len(list)):
    print(list[index])
else:
    print("Inside else block")

Naruto
Sasuke
Hinata
Ino
Inside else block


In [24]:
s=0
for i in range(1,101):
    print(i,end=" ")
    s=s+i
print("\nsum: ",s)

1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 
sum:  5050


In [19]:
s=0
for i in range(1,101):
    print(i,end=" ")
    s=s+i
    if i%10==0:
        print()
print("Sum:",s)

1 2 3 4 5 6 7 8 9 10 
11 12 13 14 15 16 17 18 19 20 
21 22 23 24 25 26 27 28 29 30 
31 32 33 34 35 36 37 38 39 40 
41 42 43 44 45 46 47 48 49 50 
51 52 53 54 55 56 57 58 59 60 
61 62 63 64 65 66 67 68 69 70 
71 72 73 74 75 76 77 78 79 80 
81 82 83 84 85 86 87 88 89 90 
91 92 93 94 95 96 97 98 99 100 
Sum: 5050


In [29]:
name=['Naruto','Sasuke','Hinata','Ino']
for x in name:
    print(x)
    if x=='Hinata':
        break

Naruto
Sasuke
Hinata


In [31]:
fruit='banana'
for x in fruit:
    if x=='n':
        break
    print(x)

b
a


In [32]:
name=['Naruto','Sasuke','Hinata','Ino']
for x in name:
    if x=='Sasuke':
        continue
    print(x)

Naruto
Hinata
Ino


In [33]:
#factorial
n=int(input("Enter number: "))
f=1
if n<0:
    print("Factorial does not exist")
if n==0:
    print("Factorial of 0! is 1")
else:
    for i in range(1,n+1):
        f*=i
    print('Factorial is: ',f)

Enter number:  5


Factorial is:  120


In [37]:
# n terms fibonacci (while loop)
n=int(input("How many terms? "))
n1,n2=1,2
print(f"{n1} \n{n2}")
count=3
while count<=n:
    n3=n1+n2
    print(n3)
    n1=n2
    n2=n3
    count+=1

How many terms?  5


1 
2
3
5
8


In [50]:
# n terms fibonacci (for loop)
a,b=1,2
n=int(input("How many terms?"))
print(a,b,end=" ")
for term in range(3,n+1):
    c=a+b
    print(c,end=' ')
    a,b=b,c
    

How many terms? 5


1 2 3 5 8 

In [27]:
# n terms fibonacci (for loop)
a,b=0,1
n=int(input("How many terms?"))
print(a,b,end=" ")
total=a+b

for term in range(3,n+1):
    c=a+b
    print(c,end=' ')
    total+=c
    a,b=b,c
print("\nSum of terms: ",total)

How many terms? 5


0 1 1 2 3 
Sum of terms:  7


In [35]:
n=int(input("Enter no of terms: "))
n1,n2=0,1
print("Fibonacci series is: ")
for i in range(n):
    print(n1,end=" ")
    n1,n2=n2,n1+n2

Enter no of terms:  5


Fibonacci series is: 
0 1 1 2 3 

Dry run for fibonacci sum

In [ ]:
# Dry Run Example → n = 5

# Step 1: a = 1, b = 2

# Step 2: n = 5

# Step 3: Prints → 1 2

# Step 4: total = a + b = 1 + 2 = 3

# Loop: for term in range(3, 6): → term = 3, 4, 5

# Iteration 1 (term = 3):
# c = a + b = 1 + 2 = 3
# Print → 3
# total = 3 + 3 = 6
# Update → a = 2, b = 3

# Iteration 2 (term = 4):
# c = a + b = 2 + 3 = 5
# Print → 5
# total = 6 + 5 = 11
# Update → a = 3, b = 5

# Iteration 3 (term = 5):
# c = a + b = 3 + 5 = 8
# Print → 8
# total = 11 + 8 = 19
# Update → a = 5, b = 8

In [43]:
#armstrong number
lower=100
upper=999
for n in range(lower,upper+1):
    temp=n
    sum=0
    while temp>0:
        digit=temp%10
        sum+=digit**3
        temp//=10 
    if n==sum:
        print(n)

153
370
371
407


In [38]:
n=int(input("Enter number: "))
sum=0
temp=n
while temp>0:
    digit=temp%10
    sum+=digit**3
    temp//=10

if sum==n:
    print(n,"is an armstrong number")
else:
    print(n,"is not an armstrong number")

Enter number:  153


153 is an armstrong number


In [44]:
lower=100
upper=999
for n in range(lower,upper+1):
    temp=n
    sum=0
    while(temp>0):
        digit=temp%10
        sum+=digit**3
        temp//=10
    if n==sum:
        print(n)
        

153
370
371
407


In [48]:
# factors of given number
n=int(input("Enter a number: "))
print("factors of",n,"are: ")
for i in range(1,n+1):
    if n%i==0:
        print(i,end=' ')
    

Enter a number:  144


factors of 144 are: 
1 2 3 4 6 8 9 12 16 18 24 36 48 72 144 

In [25]:
#prime numbers in 1 to 100
print("Prime numbers between 1 to 10 are: \n")
for n in range(1,11):
    if n>1:
        for i in range(2,n):
            if(n%i)==0:
                break
        else:
            print("Anil=",n)

Prime numbers between 1 to 10 are: 

Anil= 2
Anil= 3
Anil= 5
Anil= 7


In [4]:
for n in range(1,11):
    if n>1:
        for i in range(2,n):
            if n%i==0:
                break
        else:
            print(n)
        

2
3
5
7


In [42]:
#prime number checking
num=int(input("Enter number to check: "))
if num>1:
    for i in range(2,num):
        if num%i==0:
            print(num,"is not a prime number")
            break
    else:
        print(num," is a prime number")
else:
    print(num,"is not a prime number")

Enter number to check:  35


35 is not a prime number


In [58]:
lower_value=int(input("Enter lower bounding value "))
upper_value=int(input("Enter upper bounding value "))
print(f"Prime numbers between {lower_value} and {upper_value}:")
for n in range(lower_value,upper_value+1):
    if n>1:
        for i in range(2,n):
            if n%i==0:
                break
        else:
            print(n)
        

Enter lower bounding value  2
Enter upper bounding value  20


Prime numbers between 2 and 20:
2
3
5
7
11
13
17
19


In [40]:
#palindrome number
n=int(input("Enter number to check palindrome:"))
temp=n
rev=0
while temp>0:
    rev=rev*10+temp%10
    temp//=10

if rev==n:
    print(n,"is a palindrome.")
else:
    print(n,"is not a palindrome.")

Enter number to check palindrome: 515


515 is a palindrome.


In [5]:
#prime factors of number
n=int(input("Enter a number: "))
print(f"Prime factors of {n} are: ",end=" ")
i=2
while i<=n:
    if n%i==0:
        print(i,end=' ')
        n=n//i
    else:
        i+=1

Enter a number:  3


Prime factors of 3 are:  3 

In [6]:
#prime factors
n=int(input("Enter numbers to factorise:"))
i=2
while i<=n:
    if n%i==0:
        print(i,end=' ')
        n//=i
    else:
        i+=1

Enter numbers to factorise: 12


2 2 3 

In [45]:
#factorial
num=int(input("Enter number to find factorial: "))
fact=1
for i in range(1,num+1):
    fact*=i
print(fact)

Enter number to find factorial:  5


120


In [52]:
num=int(input("Enter a number: "))
temp=num
sum=0
while temp>0:
    sum+=temp%10
    temp//=10
print(f"Sum of digits of {num} is {sum}")

Enter a number:  123


Sum of digits of 123 is 6


In [58]:
###### perfect numbers
num=int(input("Enter number to find perfect number: "))
sum=0
for i in range(1,num):
    if num%i==0:
        sum+=i
if sum==num:
    print(num,"is a perfect number")
else:
    print(num,"is not a perfect number")

Enter number to find perfect number:  28


28 is a perfect number


In [59]:
##number pyramid
rows = int(input("Enter number of rows: "))

for i in range(1, rows + 1):
    for j in range(1, i + 1):
        print(j, end=" ")
    print()

Enter number of rows:  5


1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


In [57]:
###### reverse number
n=123
rev=0
while n>0:
    digit=n%10
    rev=rev*10+digit
    n//=10
print(rev)

321


In [48]:
#triangular pattern of stars
r=1
while r<=5:
    c=1
    while c<=r:
        print('*',end='')
        c+=1
    print()
    r+=1



*
**
***
****
*****


In [15]:
for i in range(5+1,0,-1):
    print("*"*i)

******
*****
****
***
**
*


In [17]:
rows=int(input("Enter numbers of rows:"))
for i in range(1,rows+1):
    print(' '*(rows-i),i*'*')

Enter numbers of rows: 4


    *
   **
  ***
 ****


In [10]:
r=1
while r<=5:
    c=5
    while c>=r:
        print("*",end=" ")
        c-=1
    print()
    r+=1

* * * * * 
* * * * 
* * * 
* * 
* 


In [13]:
#triangular pattern of stars in decreasing order
r=1
while r<=5:
    c=5
    while c>=r:
        print('*',end='')
        c-=1
    print()
    r+=1

*****
****
***
**
*


In [12]:
i=1
while i<=5:
    print(i*"*")
    i+=1

i=4
while i>=1:
    print(i*"*")
    i-=1

*
**
***
****
*****
****
***
**
*


In [14]:
i=1
while i<=5:
    print('*'*i)
    i+=1

#lower half
i=4
while i>=1:
    print('*'*i)
    i-=1

*
**
***
****
*****
****
***
**
*


In [26]:
#pyramid structure
n=5
for i in range(1,n+1):
    print(' '*(n-i),end='')
    print('*'*(2*i-1))

    *
   ***
  *****
 *******
*********


In [20]:
rows=int(input("Enter numbers of rows: "))
for i in range(rows+1):
    print(' '*(rows-i),'*'*(i*2-1))

Enter numbers of rows:  5


      
     *
    ***
   *****
  *******
 *********


In [22]:
rows=int(input("Enter numbers of rows: "))
for i in range(rows,0,-1):
    print(' '*(rows-i),'*'*(i*2-1))

Enter numbers of rows:  5


 *********
  *******
   *****
    ***
     *


In [26]:
rows=int(input("Enter number of rows:"))
for i in range(rows):
    print(' '*(rows-i),(i*2-1)*'*')
for i in range(rows,0,-1):
    print(' '*(rows-i),(i*2-1)*'*')

Enter number of rows: 5


      
     *
    ***
   *****
  *******
 *********
  *******
   *****
    ***
     *


In [27]:
#diamond structure
n=5
for i in range(1,n+1):
    print(' '*(n-i)+'*'*(2*i-1))
for i in range(n-1,0,-1):
    print(' '*(n-i)+'*'*(2*i-1))

    *
   ***
  *****
 *******
*********
 *******
  *****
   ***
    *


In [16]:
n=2
for i in range(2,n):
    if(n%i)==0:
        print("n=",n)
        break
else:
    print(n,end=' ')

2 

In [8]:
s='pythonstring'
s='G'+s[1:]
s


'Gythonstring'